tests/test_bibliotheca.py

```diff
@@ -1063,11 +1039,13 @@ def test_handle_event(self):
         eq_(1, pool.licenses_owned)
         eq_(1, pool.licenses_available)
 
-        # Three analytics events were collected: one for the license add
+        # Four analytics events were collected: one for the license add
         # event itself, one for the 'checkin' that made the new
-        # license available, and one for the first appearance of a new
-        # LicensePool.
-        eq_(3, analytics.count)
+        # license available, one for the first appearance of a new
+        # LicensePool, and a redundant 'license add' event
+        # which was registered with analytics but which did not
+        # affect the counts.
+        eq_(4, analytics.count)
```

Comment:
I believe 4 analytics events were always issued, but one of them went to the wrong place because when BibliothecaEventMonitor created a BibliothecaBibliothecaCoverageProvider, it didn't propagate a ReplacementPolicy that included the mock analytics. A real analytics object was used instead.

I noticed this because one of my changes made it so 2 of the analytics events are generated by the monitor and 2 are generated by the BibliothecaBibliographicCoverageProvider (through its call to bibliographic_lookup), rather than the 3/1 split I believe we had previously.

The fourth analytics event doesn't actually change anything -- by the time it makes it to update_availability_from_delta, it's out of date, but we decide to ignore it but log it anyway for the sake of completeness.

Reply:
I'm not sure it's a good idea to log the redundant event - it could be confusing when trying to analyze the analytics events.

Reply:
I filed https://jira.nypl.org/browse/SIMPLY-1484, comment on there and let me know if you agree with the rationale.